spec/features/add_employer_spec.rb

```diff
@@ -0,0 +1,46 @@
+require "rails_helper"
+
+RSpec.describe "adding an employer" do
+  context "valid inputs" do
+    it "allows a user to create an employer with name and eNapsa account number" do
+      visit new_employer_path
+      fill_in "Name", with: "My Company"
+      fill_in "Account number", with: 1234567
+      click_on "Create Employer"
+      expect(page).to have_content("My Company")
+      expect(page).to have_content(1234567)
+    end
+  end
+
+  context "with invlaid input: no account number" do
+    it "does not allow a user to create an employer without an eNapsa account number" do
+      visit new_employer_path
+      fill_in "Name", with: "My Company"
+      click_on "Create Employer"
+      expect(page).not_to have_content("My Company")
+    end
+
+    it "rerenders the new employer form and dispalys an error message if an invalid or no eNapsa account number is supplied" do
+      visit new_employer_path
+      fill_in "Name", with: "My Company"
+      click_on "Create Employer"
+      expect(page).to have_content("Account number is the wrong length (should be 7 characters)")
+    end
+  end
+
+  context "with invlaid input: no name" do
+    it "does not allow a user to create an employer without an name" do
+      visit new_employer_path
+      fill_in "Account number", with: 1234567
+      click_on "Create Employer"
+      expect(page).not_to have_content("1234567")
+    end
+
+    it "rerenders the new employer form and dispalys an error message if no name is supplied" do
+      visit new_employer_path
+      fill_in "Account number", with: 1234567
+      click_on "Create Employer"
+      expect(page).to have_content("Name can't be blank")
+    end
+  end
```

Comment:
Once you have some more granular unit tests for validations at the model layer, I would probably test fewer things at this level. 

If we think of the [testing pyramid](https://martinfowler.com/articles/practical-test-pyramid.html), we tend to test fewer things at the higher level (integration tests).

It is probably ok to have 2 test cases (i.e one for submitting a valid form, and one for submitting an invalid form).